app/src/ui/preferences/custom-integration-form.tsx

```diff
@@ -0,0 +1,209 @@
+  checkTargetPathArgument,
+  validateCustomIntegrationPath,
+  parseCustomIntegrationArguments,
+  TargetPathArgument,
+} from '../../lib/custom-integration'
+
+interface ICustomIntegrationFormProps {
+  /** ID used to prefix the IDs of some child elements */
+  readonly id: string
+  readonly path: string
+  readonly arguments: string
+  readonly onPathChanged: (path: string, bundleID?: string) => void
+  readonly onArgumentsChanged: (args: ReadonlyArray<string>) => void
+}
+
+interface ICustomIntegrationFormState {
+  readonly path: string
+  readonly arguments: string
+  /** Whether or not the current path is valid */
+  readonly isValidPath: boolean
+  /** Whether or not to show a warning for an invalid path */
+  readonly showNonValidPathWarning: boolean
+  /** Whether or not the current arguments are valid */
+  readonly isValidArgs: boolean
+  /** Whether or not to show an error for invalid arguments */
+  readonly showNonValidArgsError: boolean
+  /**
+   * Whether or not to show an error for missing target path placeholder among
+   * the arguments.
+   */
+  readonly showNoRepoPathArgError: boolean
+}
+
+/** A form for configuring a custom integration, with a path and arguments. */
+export class CustomIntegrationForm extends React.Component<
+  ICustomIntegrationFormProps,
+  ICustomIntegrationFormState
+> {
+  private pathInputRef = React.createRef<TextBox>()
+
+  public constructor(props: ICustomIntegrationFormProps) {
+    super(props)
+
+    this.state = {
+      path: props.path,
+      arguments: props.arguments,
+      isValidPath: false,
+      showNonValidPathWarning: false,
+      isValidArgs: false,
+      showNonValidArgsError: false,
+      showNoRepoPathArgError: false,
+    }
+  }
+
+  /** Focuses the path text box. */
+  public focus() {
+    this.pathInputRef.current?.focus()
+  }
+
+  public render() {
+    return (
+      <div className="custom-integration-form-container">
+        <div className="custom-integration-form-path-container">
+          <TextBox
+            label="Path"
+            value={this.state.path}
+            ref={this.pathInputRef}
+            onValueChanged={this.onPathChanged}
+            placeholder="Path to executable"
+            ariaDescribedBy={`${this.props.id}-custom-integration-path-error`}
+          />
+          <Button onClick={this.onChoosePath}>Choose…</Button>
+        </div>
+        {this.renderPathErrors()}
+        <TextBox
+          label="Arguments"
+          value={this.state.arguments}
+          onValueChanged={this.onParamsChanged}
+          placeholder="Command line arguments"
+          ariaDescribedBy={`${this.props.id}-custom-integration-args-error`}
+        />
+        {this.renderArgsErrors()}
+      </div>
+    )
+  }
+
+  private renderPathErrors() {
+    if (this.state.isValidPath || !this.state.showNonValidPathWarning) {
+      return null
+    }
+
+    const errorDescription =
+      'This directory does not appear to be a valid executable.'
```

Comment:
@sergiou87 doesn't it make more sense for the error to be: `This path does not appear to be a valid executable.`?

Reply:
🤦‍♂️  Yes, thank you! 🙇‍♂️

Reply:
Fixed in [8915043](https://github.com/desktop/desktop/pull/18954/commits/8915043c76e9654a962e96b5091ee5c22e53a90a), thanks again 💖